x/distribution/simulation/invariants.go

```diff
@@ -59,24 +59,30 @@ func CanWithdrawInvariant(k distr.Keeper, sk types.StakingKeeper) sdk.Invariant
 
 		var remaining sdk.DecCoins
 
+		valDelegationAddrs := make(map[string][]sdk.AccAddress)
+		for _, del := range sk.GetAllSDKDelegations(ctx) {
+			valAddr := del.GetValidatorAddr().String()
+			valDelegationAddrs[valAddr] = append(valDelegationAddrs[valAddr], del.GetDelegatorAddr())
+		}
+
 		// iterate over all validators
 		sk.IterateValidators(ctx, func(_ int64, val sdk.Validator) (stop bool) {
```

Comment:
alternatively, can we just loop over `valDelegationAddrs`?

Reply:
that should be fine - you do need to iterate over all validators to withdraw commission separately though

Reply:
In that case, the current iteration should suffice.